platform/default/online_file_source.cpp

```diff
@@ -176,6 +162,49 @@ class OnlineFileSource::Impl {
         }
     }
 
+
+    struct PendingRequests {
+        PendingRequests() : queue(), num_high_priority_requests(0) {}
+
+        std::list<OnlineFileRequest*> queue;
+        int num_high_priority_requests;
+
+        void remove(OnlineFileRequest* request) {
+            auto it = std::find(queue.begin(), queue.end(), request);
+            if (it != queue.end()) {
+                queue.erase(it);
+                if (!request->resource.isLowPriority) num_high_priority_requests--;
+            }
+        }
+
+        void insert(OnlineFileRequest* request) {
+            if (request->resource.isLowPriority) queue.insert(queue.end(), request);
+            else {
+                assert(num_high_priority_requests <= (int) queue.size());
+                auto new_pos = queue.begin();
+                for (int i = 0; i < num_high_priority_requests; i++) new_pos++;
```

Comment:
Instead of this iterator increment - you could replace `num_high_priority_requests` with the iterator to the last inserted high priority request. Regular insertions can then be made without traversing the list. When removing/popping requests, an additional check is needed before the removal/pop to ensure that the `lastHighPriorityRequest` iterator points to a valid list entry.